com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/parser/UriParametersParser.java

```diff
@@ -157,49 +174,70 @@ private void addHostBasedUris(Set<String> uris, String domain, List<String> host
         }
     }
 
-    private List<String> getUris(List<Map<String, Object>> parametersList) {
+    public List<String> getApplicationRoutes(List<Map<String, Object>> parametersList) {
 
-        List<String> routes = getApplicationRoutes(parametersList);
-        if (!routes.isEmpty()) {
-            return routes;
+        List<Map<String, Object>> routesMaps = RoutesValidator
+            .applyRoutesType(PropertiesUtil.getPropertyValue(parametersList, SupportedParameters.ROUTES, null));
+
+        List<String> allIndividualRoutes = CollectionUtils.emptyIfNull(routesMaps)
+            .stream()
+            .map(routesMap -> (String) routesMap.get(SupportedParameters.ROUTE))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+        if (modifyRoute) {
+            return allIndividualRoutes.stream()
+                .map(foundRoute -> modifyUri(foundRoute, parametersList))
+                .collect(Collectors.toList());
         }
 
-        List<String> hosts = getApplicationHosts(parametersList);
-        List<String> domains = getApplicationDomains(parametersList);
-        List<Integer> ports = getApplicationPorts(parametersList);
+        return allIndividualRoutes;
+    }
+
+    public String modifyUri(String inputURI, List<Map<String, Object>> parameterList) {
+        ApplicationURI modifiedURI = new ApplicationURI(inputURI);
 
-        if (domains.isEmpty() && hosts.isEmpty()) {
-            return new ArrayList<>();
+        List<String> domains = getDomainValues(parameterList);
+        List<String> hosts = getHostValues(parameterList);
+        List<Integer> ports = getPortValues(parameterList);
+
+        if (!domains.isEmpty()) {
+            modifiedURI.setDomain(domains.get(0));
```

Comment:
Why do we need to do this? A similar thing is being done in `MtaDescriptorPropertiesResolver` in the same change. Will this be necessary after https://github.com/cloudfoundry-incubator/multiapps-controller/pull/408.